dffml/model/model.py

```diff
@@ -84,6 +106,208 @@ def __call__(self) -> ModelContext:
         self._make_config_location()
         return self.CONTEXT(self)
 
+    async def __aenter__(self):
+        if self.config.location.is_file():
+            temp_dir = self._get_directory()
+            await self._run_operation(
+                "extract", self.config.location, temp_dir
+            )
+            self.location = temp_dir
+
+    async def __aexit__(self, exc_type, exc_value, traceback):
+        if self.config.location.is_file():
+            config_path = self.location / "config.json"
+            config_path.write_text(json.dumps(export(self.config)))
+            await self._run_operation(
+                "archive", self.temp_dir, self.config.location
+            )
+            shutil.rmtree(self.temp_dir)
```

Comment:
Let's make sure that we remove the tempdir if we created it, even if location is no longer a file

Reply:
okay, would push this in a future commit

Reply:
Done